Spring-Boot/Spring-Boot-Student/src/test/java/tw/waterball/judgegirl/springboot/student/controllers/it/StudentControllerIT.java

```diff
@@ -0,0 +1,173 @@
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+import static tw.waterball.judgegirl.springboot.student.view.StudentView.toViewModel;
+
+
+/**
+ * @author chaoyulee chaoyu2330@gmail.com
+ */
+@ActiveProfiles(Profiles.JWT)
+@ContextConfiguration(classes = SpringBootStudentApplication.class)
+public class StudentControllerIT extends AbstractSpringBootTest {
+    private Student student;
+
+    @Autowired
+    private JpaStudentDataPort studentRepository;
+
+    @Autowired
+    private TokenService tokenService;
+
+    @BeforeEach
+    void setup() {
+        student = new Student("name", "email@example.com", "password");
+    }
+
+    @AfterEach
+    void cleanUp() {
+        studentRepository.deleteAll();
+    }
+
+
+    private ResultActions signUp(Student student) throws Exception {
+        return mockMvc.perform(post("/api/students/signUp")
+                .contentType(MediaType.APPLICATION_JSON)
+                .content(toJson(student)));
+    }
+
+    private StudentView signUpAndGetResponseBody(Student student) throws Exception {
+        return getBody(signUp(student).andExpect(status().isOk()), StudentView.class);
+    }
+
+    private ResultActions signIn(String email, String password) throws Exception {
+        return mockMvc.perform(post("/api/students/login")
+                .contentType(MediaType.APPLICATION_JSON)
+                .content(toJson(new SignInUseCase.Request(email, password))));
+    }
+
+    private LoginResponse signInAndGetResponseBody(String email, String password) throws Exception {
+        return getBody(signIn(email, password).andExpect(status().isOk()), LoginResponse.class);
+    }
+
+    private ResultActions getStudent(Integer id, String tokenString) throws Exception {
+        return mockMvc.perform(get("/api/students/" + id)
+                .header("Authorization", "bearer " + tokenString));
+    }
+
+    private ResultActions auth(String tokenString) throws Exception {
+        return mockMvc.perform(post("/api/students/auth")
+                .header("Authorization", "bearer " + tokenString));
+    }
+
+    private LoginResponse authAndGetResponseBody(String tokenString) throws Exception {
+        return getBody(mockMvc.perform(post("/api/students/auth")
+                .header("Authorization", "bearer " + tokenString))
+                .andExpect(status().isOk()), LoginResponse.class);
+    }
+
+    @Test
+    void WhenSignUpCorrectly_ShouldReturnStudentView() throws Exception {
+        StudentView body = signUpAndGetResponseBody(student);
+        student.setId(body.getId());
+        assertEquals(toViewModel(student), body);
+    }
+
+    @Test
+    void GivenSignUpAccount_WhenLoginCorrectly_ShouldReturnLoginResponse() throws Exception {
```

Comment:
rename to `GivenSignUpAccount_WhenLoginCorrectly_ShouldReturnLoginResponseWithCorrectToken`